src/magic.h

```diff
@@ -81,6 +82,50 @@ struct fake_spell {
         max_level( max_level ), self( hit_self ) {}
 };
 
+class ter_furn_transform
+{
+
+    private:
+
+        std::string fail_message;
+
+        std::map<ter_id, weighted_int_list<ter_id>> ter_transform;
+        std::map<std::string, weighted_int_list<ter_id>> ter_flag_transform;
+        // the value is message and whether that message is good or bad. defaults to good
+        std::map<ter_id, std::pair<std::string, bool>> ter_message;
+        std::map<std::string, std::pair<std::string, bool>> ter_flag_message;
+
+        std::map<furn_id, weighted_int_list<furn_id>> furn_transform;
+        std::map<std::string, weighted_int_list<furn_id>> furn_flag_transform;
+        std::map<furn_id, std::pair<std::string, bool>> furn_message;
+        std::map<std::string, std::pair<std::string, bool>> furn_flag_message;
+
+        ter_id next_ter( const ter_id ter ) const;
+        ter_id next_ter( const std::string &flag ) const;
+        cata::optional<furn_id> next_furn( const furn_id furn ) const;
+        cata::optional<furn_id> next_furn( const std::string &flag ) const;
+
+    public:
+
+        ter_furn_transform_id id;
+        bool was_loaded;
+
+        void ter_add_message( const Creature &critter, const tripoint &location, const ter_id ter ) const;
+        void furn_add_message( const Creature &critter, const tripoint &location,
+                               const furn_id furn ) const;
+        void transform( const tripoint &pos ) const;
+
+        static void load_transform( JsonObject &jo, const std::string &src );
+        void load( JsonObject &jo, const std::string & );
+        /**
+         * All spells in the game.
```

Comment:
Uhm, doesn't it return lots of `ter_furn_transform` objects? Does it also return "all the spells in the game" along?